/**
* @license Apache-2.0
*
* Copyright (c) 2022 The Stdlib Authors.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*    http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/

import hoursInMonth = require( './index' );


// TESTS //

// The function returns a number...
{
	hoursInMonth(); // $ExpectType number
	hoursInMonth( 2 ); // $ExpectType number
	hoursInMonth( 'February' ); // $ExpectType number
	hoursInMonth( new Date() ); // $ExpectType number
}

// The function does not compile if provided a value other than a number, string, or Date for the first argument...
{
	hoursInMonth( true ); // $ExpectError
	hoursInMonth( false ); // $ExpectError
	hoursInMonth( null ); // $ExpectError
	hoursInMonth( [] ); // $ExpectError
	hoursInMonth( {} ); // $ExpectError
	hoursInMonth( ( x: number ): number => x ); // $ExpectError
}

// The function does not compile if provided a value other than a number as the second argument...
{
	hoursInMonth( 2, true ); // $ExpectError
	hoursInMonth( 2, false ); // $ExpectError
	hoursInMonth( 2, 'beep' ); // $ExpectError
	hoursInMonth( 2, [] ); // $ExpectError
	hoursInMonth( 2, {} ); // $ExpectError
	hoursInMonth( 2, ( x: number ): number => x ); // $ExpectError
}

// The compiler throws an error if the function is provided an unsupported number of arguments...
{
	hoursInMonth( 2, 2, 2 ); // $ExpectError
}

